app/javascript/components/common/ExerciseWidget.tsx

```diff
@@ -72,50 +76,91 @@ const ReferenceElement = forwardRef<
     onMouseEnter?: () => void
     onMouseLeave?: () => void
   }
->(({ exercise, track, solution, size, ...props }, ref) => {
-  if (solution) {
-    return (
-      <a
-        ref={ref as React.RefObject<HTMLAnchorElement>}
-        href={solution.url}
-        className={`c-exercise-widget --${solution.status} --${size}`}
-        {...props}
-      >
-        <ExerciseIcon iconUrl={exercise.iconUrl} title={exercise.title} />
-        <Info exercise={exercise} solution={solution} track={track} />
-        <GraphicalIcon icon="chevron-right" className="--action-icon" />
-      </a>
-    )
-  } else if (exercise.isUnlocked) {
-    const classNames = [
-      'c-exercise-widget',
-      '--available',
-      `--${size}`,
-      exercise.isRecommended ? '--recommended' : '',
-    ].filter((name) => name.length > 0)
-    return (
-      <a
-        href={exercise.links.self}
-        className={classNames.join(' ')}
-        ref={ref as React.RefObject<HTMLAnchorElement>}
-        {...props}
-      >
-        <ExerciseIcon iconUrl={exercise.iconUrl} title={exercise.title} />
-        <Info exercise={exercise} solution={solution} track={track} />
-        <GraphicalIcon icon="chevron-right" className="--action-icon" />
-      </a>
-    )
-  } else {
-    return (
-      <div
-        className={`c-exercise-widget --locked --${size}`}
-        ref={ref as React.RefObject<HTMLDivElement>}
-        {...props}
-      >
-        <ExerciseIcon iconUrl={exercise.iconUrl} title={exercise.title} />
-        <Info exercise={exercise} solution={solution} track={track} />
-        <GraphicalIcon icon="lock" className="--action-icon" />
-      </div>
+>(
+  (
+    {
+      exercise,
+      track,
+      solution,
+      renderAsLink,
+      renderBlurb,
+      isSkinny,
+      ...props
+    },
+    ref
+  ) => {
+    const info = (
+      <Info
+        exercise={exercise}
+        solution={solution}
+        track={track}
+        renderBlurb={renderBlurb}
+        isSkinny={isSkinny}
+      />
     )
+    if (solution || exercise.isUnlocked) {
+      const classNames = [
+        'c-exercise-widget',
+        `--${solution ? solution.status : 'available'}`,
+        exercise.isRecommended ? '--recommended' : '',
+        `--${renderAsLink ? 'interactive' : 'static'}`,
+        isSkinny ? '--skinny' : '',
+      ]
```

Comment:
You could consider indenting by two spaces, to have the closing of the array and the `.filter` be at the same indentation, helping reinforce that they belong together.

```suggestion
          'c-exercise-widget',
          `--${solution ? solution.status : 'available'}`,
          exercise.isRecommended ? '--recommended' : '',
          `--${renderAsLink ? 'interactive' : 'static'}`,
          isSkinny ? '--skinny' : '',
        ]
```

Reply:
I'll check but I feel like prettier (or whatever it is) undoes that.

Reply:
Aha! Weirdness

Reply:
Yeah, I've tried. It undoes it 🤷